src/GovUk.Education.ExploreEducationStatistics.Admin/Services/UpdateChartFilesService.cs

```diff
@@ -0,0 +1,168 @@
+
+                        await _contentDbContext.SaveChangesAsync();
+
+                        _logger.LogInformation(
+                            $"Update blob: {fileReference.ReleaseId}:{fileReference.Filename} successfully");
+                    }
+                }
+            }
+
+            return true;
+        }
+
+        private async Task<bool> CopyBlob(ReleaseFileReference fileReference, CloudBlobContainer blobContainer, string oldBlobPath, Guid newId)
+        {
+            var newBlobPath = AdminReleasePathWithFilename(fileReference.ReleaseId, newId.ToString());
+
+            var sourceBlob = blobContainer.GetBlockBlobReference(oldBlobPath);
+            var targetBlob = blobContainer.GetBlockBlobReference(newBlobPath);
+                        
+            await targetBlob.StartCopyAsync(sourceBlob);
+            targetBlob.FetchAttributes();
+            while (targetBlob.CopyState.Status == CopyStatus.Pending) {
+                await Task.Delay(1000);
+                targetBlob.FetchAttributes();
+            }
+
+            if (targetBlob.CopyState.Status != CopyStatus.Success) {
+                _logger.LogError($"Failed to copy blob: {fileReference.ReleaseId}:{fileReference.Filename}");
+                return false;
+            }
+
+            await sourceBlob.DeleteAsync();
+            
+            _logger.LogInformation($"Copied blob: {fileReference.ReleaseId}:{fileReference.Filename} successfully");
+
+            return true;
+        }
+        
+        private async Task UpdateDataBlock(ReleaseFileReference releaseFileReference, Guid newChartId)
+        {
+            var releaseIds = await _contentDbContext.ReleaseFiles
+                .Where(rf => rf.ReleaseFileReferenceId == releaseFileReference.Id)
+                .Select(rf => rf.ReleaseId)
+                .Distinct()
+                .ToListAsync();
+            
+            var blocks = GetDataBlocks(releaseIds);
+
+            foreach (var block in blocks)
+            {
+                var infoGraphicChart = block.Charts
+                    .OfType<InfographicChart>()
+                    .FirstOrDefault();
+
+                if (infoGraphicChart != null && infoGraphicChart.FileId == releaseFileReference.Filename)
+                {
+                    block.Charts.Remove(infoGraphicChart);
+                    infoGraphicChart.FileId = newChartId.ToString();
+                    block.Charts.Add(infoGraphicChart);
+                    _contentDbContext.DataBlocks.Update(block);
+                }
+            }
+        }
+        
+        private List<DataBlock> GetDataBlocks(List<Guid> releaseIds)
+        {
+            return _contentDbContext
+                .ReleaseContentBlocks
+                .Include(join => join.ContentBlock)
+                .ThenInclude(block => block.ContentSection)
+                .Where(join => releaseIds.Contains(join.ReleaseId))
```

Comment:
Param should be `releaseContentBlock` (or `rcb`) instead of `join`.

Reply:
done